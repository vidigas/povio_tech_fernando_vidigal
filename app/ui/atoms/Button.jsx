import React from 'react';
import injectSheet from 'react-jss';
import propTypes from 'prop-types';

const buttonStyle = {
  button: {
      color: 'white',
      border: 'none',
      outline: 'none',
      padding: '10px 20px',
      cursor: 'pointer',
      backgroundColor:({ isLiked }) => isLiked ? '#ff8282': '#81D2AE',
      fontWeight: 'bold',
      borderRadius: '10px',
      minWidth: '30px',
      position: 'relative',
      top: '10px',
      height:'35px',
      margin: '15px'
  },
  dislike: {
      backgroundColor: 'red'
  }
};

const Button = ({ label, onClick, className, classes, type, key, isLiked }) => {
  return (
    <button onClick={onClick} className={`${classes.button} ${className}`} type={type} key={key}>
      {label}
    </button>
  );
}

Button.propTypes = {
	onClick: propTypes.func.isRequired,
	label: propTypes.string,
	className: propTypes.string,
	type: propTypes.string
};

Button.defaultProps = {
	type: 'button',
	className: '',
	label: ''
};

export default injectSheet(buttonStyle)(Button);