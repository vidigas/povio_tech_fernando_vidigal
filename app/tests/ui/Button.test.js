import React from 'react';
import renderer from 'react-test-renderer';
import Button from '../../ui/atoms/Button';

const func = () => {};

it('renders correctly', () => {
  const tree = renderer
    .create(<Button label="some label" onClick={func} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});