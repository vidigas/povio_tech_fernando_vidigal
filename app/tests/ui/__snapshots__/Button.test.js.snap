// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<button
  className="Button-button-0-1-1 Button-button-0-1-3 "
  onClick={[Function]}
  type="button"
>
  some label
</button>
`;
